File: frontend/packages/types/src/branch.ts
export type BranchStatus = "active" | "inactive";

export interface Branch {
  id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  status: BranchStatus;
  createdAt?: string;
}

export type CreateBranchPayload = {
  name: string;
  address: string;
  lat: number;
  lng: number;
  status?: BranchStatus;
};

export type UpdateBranchPayload = Partial<CreateBranchPayload>;

File: backend/src/main/java/com/backend/bikerental/entity/Branch.java
package com.backend.bikerental.entity;

import com.backend.bikerental.enums.BranchStatus;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "branches")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Branch {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(length=36)
    String id;
    String name;
    String address;
    @Column(precision = 10, scale = 8)
    BigDecimal lat;
    @Column(precision = 11, scale = 8)
    BigDecimal lng;
    @Enumerated(EnumType.STRING)
    BranchStatus status;
    @CreationTimestamp
    @Column(updatable = false)
    LocalDateTime createdAt;
    @UpdateTimestamp
    LocalDateTime updatedAt;

}
